util/benchmark/ec/p256_awslc_ossl/src/benchmark_ecdh.c

```diff
@@ -0,0 +1,183 @@
+/*
+------------------------------------------------------------------------------------
+ Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ SPDX-License-Identifier: Apache-2.0
+------------------------------------------------------------------------------------
+*/
+
+#include <openssl/evp.h>
+#include <openssl/ec.h>
+#ifdef AWSLC_BENCHMARK
+#include <openssl/mem.h>
+#endif
+
+#include "benchmark.h"
+
+#define MAX_ECDH_SIZE   256
+
+void benchmark_ecdh_p256(uint64_t msec)
+{
+    EVP_PKEY_CTX *kctx = NULL;
+    EVP_PKEY_CTX *test_ctx = NULL;
+    EVP_PKEY_CTX *ctx = NULL;
+    EVP_PKEY_CTX *pctx = NULL;
+    EVP_PKEY *params = NULL;
+    EVP_PKEY *key_A = NULL;
+    EVP_PKEY *key_B = NULL;
+    unsigned char *secret_a = NULL;
+    unsigned char *secret_b = NULL;
+    size_t outlen;
+    size_t test_outlen;
+    int ecdh_checks = 1;
+    uint64_t start, end, num_itr;
+    uint64_t usec = msec * 1000;
+
+    if (
```

Comment:
Not for this PR, but maybe in the future we want to have something like "GUARD" macros in s2n (https://github.com/awslabs/s2n/blob/main/utils/s2n_safety.h#L185). That way we would be able to avoid these big if statements.

Reply:
I know I know, it's a matter of taste I guess. GUARD macros print the name of the failing function IIRC so you know where exactly you failed and also abort the flow on fail. Anyway, it's not important for this PR so I'll resolve the comment.